source/code/plugin/in_containerinventory.rb

```diff
@@ -0,0 +1,250 @@
+            end
+          end
+          instance['StartedTime'] = stateValue['StartedAt']
+          instance['FinishedTime'] = stateValue['FinishedAt']
+        else
+          $log.info("Attempt in ObtainContainerState to get container: #{container['Id']} state information returned null")
+        end
+        rescue => errorStr
+          $log.warn("Exception in obtainContainerState: #{errorStr}")
+      end
+    end
+
+    def obtainContainerHostConfig(instance, container)
+      begin
+        hostConfig = container['HostConfig']
+        if !hostConfig.nil?
+          links = hostConfig['Links']
+          instance['Links'] = ""
+          if !links.nil?
+            linksString = links.to_s
+            instance['Links'] = (linksString == "null")? "" : linksString
+          end
+          portBindings = hostConfig['PortBindings']
+          instance['Ports'] = ""
+          if !portBindings.nil?
+            portBindingsString = portBindings.to_s
+            instance['Ports'] = (portBindingsString == "null")? "" : portBindingsString
+          end
+        else
+          $log.info("Attempt in ObtainContainerHostConfig to get container: #{container['Id']} host config information returned null")
+        end
+        rescue => errorStr
+          $log.warn("Exception in obtainContainerHostConfig: #{errorStr}")
+        end
+    end
+
+    def inspectContainer(id, nameMap)
+      containerInstance = {}
+      begin
+        container = DockerApiClient.dockerInspectContainer(id)
+        if !container.nil? && !container.empty?
+          containerInstance['InstanceID'] = container['Id']
+          containerInstance['CreatedTime'] = container['Created']
+          containerName = container['Name']
+          if !containerName.nil? && !containerName.empty?
+            # Remove the leading / from the name if it exists (this is an API issue)
+            containerInstance['ElementName'] = (containerName[0] == '/') ? containerName[1..-1] : containerName
+          end
+          imageValue = container['Image']
+          if !imageValue.nil? && !imageValue.empty?
+            containerInstance['ImageId'] = imageValue
+            repoImageTagArray = nameMap[imageValue]
+            if nameMap.has_key? imageValue
+              containerInstance['Repository'] = repoImageTagArray[0]
+              containerInstance['Image'] = repoImageTagArray[1]
+              containerInstance['ImageTag'] = repoImageTagArray[2]
+            end
+          end
+          obtainContainerConfig(containerInstance, container);
+          obtainContainerState(containerInstance, container);
+          obtainContainerHostConfig(containerInstance, container);
+        end
+      rescue => errorStr
+          $log.warn("Exception in inspectContainer: #{errorStr} for container: #{id}")
+      end
+      return containerInstance
+    end
+
+    def enumerate
+      currentTime = Time.now
+      emitTime = currentTime.to_f
+      batchTime = currentTime.utc.iso8601
```

Comment:
This needs to go as 'CollectionTime' (which will have to be mapped to TimeGenerated @ ingestion)

Reply:
I have fixed this in the ContainerInventory workflow